controllers/spacebindingrequest/spacebindingrequest_controller.go

```diff
@@ -92,7 +99,360 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 
-	// todo implement logic
+	if util.IsBeingDeleted(spaceBindingRequest) {
+		logger.Info("spaceBindingRequest is being deleted")
+		return reconcile.Result{}, r.ensureSpaceBindingDeletion(logger, memberClusterWithSpaceBindingRequest, spaceBindingRequest)
+	}
+	// Add the finalizer if it is not present
+	if err := r.addFinalizer(logger, memberClusterWithSpaceBindingRequest, spaceBindingRequest); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	createdOrUpdated, err := r.ensureSpaceBinding(logger, memberClusterWithSpaceBindingRequest, spaceBindingRequest)
+	// if there was an error or if spaceBinding was just created or updated,
+	// let's just return.
+	if err != nil || createdOrUpdated {
+		return ctrl.Result{}, err
+	}
+
+	// set ready condition on spaceBindingRequest
+	err = r.updateStatus(spaceBindingRequest, memberClusterWithSpaceBindingRequest, toolchainv1alpha1.Condition{
+		Type:   toolchainv1alpha1.ConditionReady,
+		Status: corev1.ConditionTrue,
+		Reason: toolchainv1alpha1.SpaceBindingRequestProvisionedReason,
+	})
 
 	return ctrl.Result{}, err
 }
+
+func (r *Reconciler) ensureSpaceBindingDeletion(logger logr.Logger, memberClusterWithSpaceBindingRequest cluster.Cluster, spaceBindingRequest *toolchainv1alpha1.SpaceBindingRequest) error {
+	logger.Info("ensure spacebinding deletion")
+	if !util.HasFinalizer(spaceBindingRequest, toolchainv1alpha1.FinalizerName) {
+		// finalizer was already removed, nothing to delete anymore...
+		return nil
+	}
+	spaceBinding, err := r.getSpaceBinding(spaceBindingRequest)
+	if err != nil {
+		return err
+	}
+	if isBeingDeleted, err := r.deleteSpaceBinding(logger, spaceBinding); err != nil {
+		return r.setStatusTerminatingFailed(logger, memberClusterWithSpaceBindingRequest, spaceBindingRequest, err)
+	} else if isBeingDeleted {
+		if err := r.setStatusTerminating(memberClusterWithSpaceBindingRequest, spaceBindingRequest); err != nil {
+			return errs.Wrap(err, "error updating status")
+		}
+		return nil
+	}
+
+	// Remove finalizer from SpaceBindingRequest
+	util.RemoveFinalizer(spaceBindingRequest, toolchainv1alpha1.FinalizerName)
+	if err := memberClusterWithSpaceBindingRequest.Client.Update(context.TODO(), spaceBindingRequest); err != nil {
+		return r.setStatusTerminatingFailed(logger, memberClusterWithSpaceBindingRequest, spaceBindingRequest, errs.Wrap(err, "failed to remove finalizer"))
+	}
+	logger.Info("removed finalizer")
+	return nil
+}
+
+// getSpaceBinding retrieves the spacebinding created by the spacebindingrequest
+func (r *Reconciler) getSpaceBinding(spaceBindingRequest *toolchainv1alpha1.SpaceBindingRequest) (*toolchainv1alpha1.SpaceBinding, error) {
+	spaceBindings := &toolchainv1alpha1.SpaceBindingList{}
+	spaceBindingLabels := runtimeclient.MatchingLabels{
+		toolchainv1alpha1.SpaceBindingRequestLabelKey:          spaceBindingRequest.GetName(),
+		toolchainv1alpha1.SpaceBindingRequestNamespaceLabelKey: spaceBindingRequest.GetNamespace(),
+	}
```

Comment:
I'm not 100% sure I understand how I would use it, since the spacebinding now is being created by this utility function https://github.com/codeready-toolchain/toolchain-common/blob/c909c72d8f627a9f72917531eded2a5e765faea1/pkg/spacebinding/spacebinding.go#L14 

Maybe I can add some utility function like `WithSpaceBindingRequestLabels` in that package and use that instead of setting the labels directly as I'm doing now when creating the object https://github.com/codeready-toolchain/host-operator/pull/848/files#diff-e48d49741db80226348ae5ffd08100dd10afa2e83e5d18b2a94776dfdfb616f3R304-R305

Reply:
never mind - I made the comment when I started reviewing the PR. When I saw the rest of the code, I agree that it doesn't make any sense.